src/parse-nm.c

```diff
@@ -250,8 +666,52 @@ netplan_parse_keyfile(const char* filename, GError** error)
                 _kf_clear_key(kf, "wifi", "mode");
         }
 
-        ap->hidden = g_key_file_get_boolean(kf, "wifi", "hidden", NULL);
-        _kf_clear_key(kf, "wifi", "hidden");
+        tmp_str = g_key_file_get_string(kf, "ipv4", "method", NULL);
+        if (tmp_str && g_strcmp0(tmp_str, "shared") == 0) {
+            ap->mode = NETPLAN_WIFI_MODE_AP;
+            _kf_clear_key(kf, "ipv4", "method");
+        }
+        g_free(tmp_str);
+
```

Comment:
Maybe a proposition regarding `netplan_parse_keyfile()` - but not for this PR, oh no, since this one is already quite big! But the proposition is to maybe divide this function into smaller ones. It's nothing bad as-is right now, just I guess most good-practice guidelines suggest for the functions to be relatively small. Guess somehow logically dividing the contents to 2-3 functions that would be called from inside could make it a bit better. Then again, this is only a suggestion for the future, since we need to think if this won't make the code even more complex than it is right now.

Reply:
ACK! :+1: